corebehrt/functional/causal/checks.py

```
@@ -0,0 +1,47 @@
+    Requires data_end, data_start, min_follow_up, min_lookback.
+    """
+    if "data_end" not in time_windows:
+        raise ValueError("data_end must be provided")
+    if "data_start" not in time_windows:
+        raise ValueError("data_start must be provided")
+
+    try:
+        data_end = pd.Timestamp(**time_windows["data_end"])
+    except KeyError as e:
+        print(e)
+        raise ValueError("data_end must be provided as year, month, day")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)